ui/app/clinical/consultation/controllers/conceptSetPageController.js

```diff
@@ -240,11 +241,36 @@ angular.module('bahmni.clinical')
                         });
                         if (currentLabel) { label = currentLabel.display; }
                     }
-                    forms.push(new Bahmni.ObservationForm(formUuid, $rootScope.currentUser,
-                        formName, formVersion, observations, label));
+                    if ($scope.isFormEditableByTheUser(observationForm)) {
+                        var tempForm = new Bahmni.ObservationForm(formUuid, $rootScope.currentUser,
+                                                                   formName, formVersion, observations, label);
+                        tempForm.privileges = privileges;
+                        forms.push(tempForm);
+                    }
                 });
+
                 return forms;
             };
+            $scope.isFormEditableByTheUser = function (form) {
+                var result = false;
+                if (form.privileges.length != 0) {
```

Comment:
@buvaneswari-arun can you also check **form.privileges** parameter is there as part of the same if condition. If the **form.privileges** is not defined then it will throw runtime error.

Reply:
Done